src/phase_space_layouts/out_channel/flat_phase_space.jl

```diff
@@ -0,0 +1,149 @@
+# here we implement RAMBO
+
+struct FlatPhaseSpaceLayout{INPSL} <: QEDcore.AbstractOutPhaseSpaceLayout{INPSL}
+    in_psl::INPSL
+end
+
+function QEDcore.phase_space_dimension(
+    proc::AbstractProcessDefinition,
+    model::AbstractModelDefinition,
+    psl::FlatPhaseSpaceLayout,
+)
+    return 4 * number_outgoing_particles(proc)
+end
+
+QEDcore.in_phase_space_layout(psl::FlatPhaseSpaceLayout) = psl.in_psl
+
+function QEDcore._build_momenta(
+    proc::AbstractProcessDefinition,
+    model::AbstractModelDefinition,
+    in_moms::Tuple,
+    psl::FlatPhaseSpaceLayout,
+    out_coords::Tuple,
+)
+    # preparing inchannel
+    Ptot = sum(in_moms)
+    boost_from_rest = inv(_unsafe_rest_boost(Ptot))
+
+    sqrt_s = Base.sqrt_llvm(Ptot * Ptot)
+    out_mass_sum = sum(mass.(outgoing_particles(proc)))
+    sqrt_s >= out_mass_sum || throw(
+        InvalidInputError(
+            """
+            sum of the masses of the outgoing particles <$out_mass_sum> must not exceed the
+            center-of-momentum energy <$sqrt_s>!
+            """,
+        ),
+    )
+
+    out_moms = _massive_rambo_moms(out_coords, sqrt_s, mass.(outgoing_particles(proc)))
+
+    return boost_from_rest.(out_moms)
+end
+
+function _massive_rambo_moms(c, ss, masses)
+    massless_moms = _massless_rambo_moms(c, ss)
+    energies = getT.(massless_moms)
+    xi = _find_scaling_factor(masses, energies, ss)
+    return _scale_rambo_moms(xi, masses, massless_moms)
+end
+
+function _to_be_solved(xi, masses, p0s, ss)
+    s = mapreduce(
+        x -> Base.sqrt_llvm(@inbounds x[1]^2 + xi^2 * x[2]^2), +, zip(masses, p0s)
+    )
+    return s - ss
+end
+
+function _find_scaling_factor(masses, energies, ss)
+    f = x -> _to_be_solved(x, masses, energies, ss)
+    xi = find_zero(f, 2, Order1())
+    return xi
+end
+
+"""
+
+massless momentum of a single particle from uniformly distributed coordinates
+"""
+function _single_rambo_mom(single_coords)
+    a, b, c, d = single_coords
+    cth = 2 * c - 1
+    sth = Base.sqrt_llvm(1 - cth^2)
+    phi = 2 * pi * d
+    p0 = -log(a) - log(b)
+    p1 = p0 * sth * cos(phi)
+    p2 = p0 * sth * sin(phi)
+    p3 = p0 * cth
+    return SFourMomentum(p0, p1, p2, p3)
+end
+
+"""
+
+build tuple of partition by four of a given tuple
+"""
+function _tuple_partition_by_four(c)
+    N = length(c)
+    m = Int(N / 4)
```

Comment:
I think an actual integer division would be more appropriate here.
```suggestion
    m = div(N, 4)
```

Reply:
done in 0538ad9